### PR TITLE
303 dpp producer reset

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description

Currently it is possible for the SO Service to enter an unrecoverable error state depending on when it sends it’s last DPP for a given green phase. This is because to confirm a DPP has been received by the TSC Service, on the SO service we confirm that the size of the current spat has increased. If we send a DPP in the final SO iteration for a given green ( time to yellow is < so frequency), the next time we check the spat, it may have the added phase but its size will seem unchanged since the previous green is now no longer present.  
## Related Issue
#303 

## Motivation and Context

Make SO service robust to edge case scenarios

## How Has This Been Tested?
NA

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: signal_opt_service/src/signal_opt_service.cpp [hunk 1]
```diff
@@ -213,11 +217,21 @@ namespace signal_opt_service
                         new_dpp_generated = false;
                     }
                     else {
-                        SPDLOG_WARN("Skipping SO iteration. Previously sent DPP is not yet reflected in SPaT!");
+                        // Increment sequential skip count for each skip
+                        sequential_skips_for_missing_dpp++;
+                        SPDLOG_WARN("Skipping SO iteration {0}. Previously sent DPP is not yet reflected in SPaT!", sequential_skips_for_missing_dpp);
+                        if (sequential_skips_for_missing_dpp > 5 ) {
```
Comment (Saeid-Solei) [inline signal_opt_service/src/signal_opt_service.cpp:223]: In our successful runs, are we sure that SO will receive the updated SPaT within next 5 iteration after sending the dpp? Do we want to make it a configurable parameter?

Comment (paulbourelly999) [reply signal_opt_service/src/signal_opt_service.cpp:223]: Yes we are sure, in fact I have never seen a missed DPP in all data collection. 5 missed DPPs would equate to approximately 2 seconds of missed information. The only behavior I have see is ~30ms latency causing a DPP to be invalid once it is received.

Comment (paulbourelly999) [reply signal_opt_service/src/signal_opt_service.cpp:223]: Additionally even though this is impossible, if we sent a DPP and it hasn't been received for 2 seconds we can likely assume it had been dropped some how or is now invalid so coming up with a new one is the appropriated response I think.

Comment (Saeid-Solei) [reply signal_opt_service/src/signal_opt_service.cpp:223]: You are saying that 5 iterations means 2 seconds of missed information. I guess this depends on the frequencies we define. I am fine with leaving 5 here for now. But we probably should be careful when changing the SO frequency.

**Pull request merged**